rosjava/src/main/java/org/ros/internal/node/topic/DefaultPublisher.java

```diff
@@ -107,9 +109,16 @@ public boolean getLatchMode() {
 
   @Override
   public void shutdown(long timeout, TimeUnit unit) {
-    signalOnShutdown(timeout, unit);
-    outgoingMessageQueue.shutdown();
-    listeners.shutdown();
+    try{
+      shutdownLatch = new CountDownLatch(listeners.size());
+      signalOnShutdown(timeout, unit);
+      shutdownLatch.await(timeout, unit);
+      outgoingMessageQueue.shutdown();
+      listeners.shutdown();
+    }
+    catch (InterruptedException e) {
```

Comment:
nit: please put `catch` statement at the same line of the closing brace above: ` } catch (InterruptedException e) {`